chainstate/src/detail/mod.rs

```diff
@@ -194,6 +194,32 @@ impl<S: BlockchainStorage, V: TransactionVerificationStrategy> Chainstate<S, V>
         }
     }
 
+    fn check_genesis(&self) -> Result<(), InitializationError> {
```

Comment:
maybe cover it with a simple unit test?

Reply:
Yes, please. I was gonna suggest this. A test at height zero, and at another arbitrary height would be great.